sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/ValuePostprocessor.scala

```diff
@@ -218,4 +220,68 @@ final class ValueEnricher(
     enrichTransaction(incompleteTx.transaction).map(transaction =>
       incompleteTx.copy(transaction = transaction)
     )
+
+  /*
+    This will renormalize a contract w.r.t a (up/down)grade template.
+    In practice this potentially
+     - adds/removes none fields
+     - change the transaction version.
+   */
+  private[this] def reversionValue(
+      typ: Ast.Type,
+      value: Value,
+      dstVer: TransactionVersion,
+  ): Result[Value] =
+    translateValue(typ, value).map(_.toNormalizedValue(dstVer))
+
+  // This
+  def reversion(
```

Comment:
I've experimented with using this API directly, and as @matthiasS-da observed, the problem we have is that we may observe several `Nodes` that bind the contract to different target template ids. As I expect we want to use the one with the maximum package version. Then we also need a method an API that calls this converter with the 

we only populate with view with a single contract instance although the transaction tree:
```
 def reversion(
      contract: FatContractInstance,
      maxVersionOfTmplId: Ref.TypeConName,
  ): Result[FatContractInstance] 
```